scripts/script-tag.js

```diff
@@ -82,33 +84,42 @@
             let productType = '';
 
             hcpreorderShipsFrom.css('visibility', 'hidden');
-            if (preorderButton.is(':input')) {
-                preorderButton.val(addToCartLabel);
-            } else {
+            if (preorderButton.is(':button')) {
                 preorderButton.html(addToCartLabel);
+            } else {
+                preorderButton.val(addToCartLabel);
             }
 
             // removing the click event with handler addToCart
             preorderButton.off('click', addToCart);
             preorderButton.siblings().css('display', 'block');
 
-            let checkItemAvailablity = await isItemAvailableForOrder().then((product) => {
-                // checking what type of tag product contains (Pre-Order / Back-order) and on the basis of that will check for metafield
-                productType = product.tags.includes('HC:Pre-Order') ? 'Pre-Order' : product.tags.includes('HC:Backorder') ? 'Back-Order' : ''
+            const currentProductMetaData = hc_metaData[variantId];
+
+            const metafieldInformation = currentProductMetaData && currentProductMetaData.hcPromiseDate ? JSON.parse(currentProductMetaData.hcPromiseDate) : '';
 
+            let checkItemAvailablity = await isItemAvailableForOrder().then((product) => {
+                if(metafieldInformation && metafieldInformation.status) {
```

Comment:
It seems the below logic is redundant for the product and category page. We should implement method for the same